File: BrahMos/AlgoMap/Arrays & Strings/2239_Find_Closest_Number_to_Zero.cpp
class Solution
{
public:
    int findClosestNumber(vector<int> &nums)
    {
        int ans = INT_MAX;
        int n = nums.size();
        int idx = 0;
        for (int i = 0; i < n; i++)
        {
            if (ans > abs(nums[i]))
            {
                ans = abs(nums[i]);
                idx = i;
            }
            else if (ans == abs(nums[i]))
            {
                if (nums[i] > nums[idx])
                {
                    idx = i;
                }
            }
        }
        return nums[idx];
    }
};

File: BrahMos/AlgoMap/Arrays & Strings/2239_Find_Closest_Number_to_Zero.py
import math 
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        mini,index,ans=sys.maxsize,0,0
        for i,num in enumerate(nums):
            temp=abs(num)
            if mini>temp:
                ans=num
                mini=temp
            elif mini==temp:
                if num>ans:
                    ans=num
        return ans

